src/solver/conv_mlir_igemm_wrw_xdlops.cpp

```diff
@@ -113,5 +114,20 @@ ConvSolution ConvMlirIgemmWrWXdlops::GetSolution(const ConvolutionContext& ctx,
 #endif
 }
 
+std::size_t ConvMlirIgemmWrWXdlops::GetWorkspaceSize(const ConvolutionContext& ctx) const
+{
+    if(ctx.IsFp32())
+        return 0;
+    else
+    {
+        const auto k = ConvolutionContextInterpreter::GetOutputChannelK(ctx);
+        const auto c = ConvolutionContextInterpreter::GetInputChannelC(ctx);
+        const auto y = ConvolutionContextInterpreter::GetFilterHeightY(ctx);
+        const auto x = ConvolutionContextInterpreter::GetFilterWidthX(ctx);
+
+        return k * c * y * x * miopen::GetTypeSize(miopenFloat);
+    }
+}
```

Comment:
`MiirGetWorkspaceSize` takes a string, instead of `ConvolutionContext`. How do I populate such a string?

In `ConvMlirIgemmWrWXdlops::GetSolution`, I can see some logic to construct `construction_parameters.comp_options)`, but it also depends on an `PerformanceConvMlirIgemmXdlops` object which isn't available here.

Reply:
@jerryyin I think I've figured out how to do it after examining the interface more clearly.

Reply:
ConvolutionContext has every information we need to construct that string. To use it, you can:

```cpp
  MiirGetWorkspaceSize(mlir::ConstructBuildOptions(ctx, true));
```

Since I can see you already setup the workspace using below, I don't think you need this `GetWorkspaceSize()`

```cpp
    result.workspace_sz = MiirGetWorkspaceSize(construction_parameters.comp_options);
```

Reply:
Pushed another commit to consolidate `MiirGetWorkspaceSize` call to a single place in `GetWorkspaceSize`.